### PR TITLE
fix: gfx1151, wrong logits

### PR DESCRIPTION
**Pull request by Error00101**
ROCm Produces very high perplexity on RDNA3.5 with promts >n_ubatch (Reproduceable with llama-perplexity)
Disabling the host buffer foregoes the corruption.

## Overview

From what i can read from the AMD Fork there is a MMQ tile-barrier race with the amd gfx115x series (host buffer to gpu).
https://github.com/AMD-Ecosystem/llama.cpp/pull/75

which then led to a bit of latency which is fixed by 
https://github.com/AMD-Ecosystem/llama.cpp/pull/72/changes/865374bbea0a65966c5d1a79c0c1f73ac8f1bfb4

Now RDNA3.5 excludes direct host access while other architectures retain it.
Eliminates the whole race condition and subsequent corruption before it even happens.

## Additional information

I did not create the fix. I just tested it localy and created this PR.

Fixes: Eval bug: HIP/ROCm on gfx1151, wrong logits (not a crash), triggered by prompts longer than n_ubatch.
https://github.com/ggml-org/llama.cpp/issues/28211

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure:  YES (Cursor)

The fix was created by https://github.com/liangliangchang

I tested the fix: ROCm now produces the same perplexity as Vulkan

### CODE REVIEW

**Comment by Error00101** [comment]
Was fixed by d4389a4dd920d24c9592f1dc3badbd69be23bd09
